src/main/java/org/spongepowered/common/mixin/core/item/MixinItemFishingRod.java

```diff
@@ -54,10 +54,12 @@ public ItemStack onItemRightClick(ItemStack itemStack, World world, EntityPlayer
             player.fishEntity.handleHookRetraction();
         } else {
             EntityFishHook fishHook = new EntityFishHook(world, player);
-            if (!Sponge.getGame().getEventManager()
+            if (world.isRemote || !Sponge.getGame().getEventManager()
```

Comment:
Honestly, it seems a bit confusing to fire events on the client thread at all. For example, if you run this code on the client:
```java
import org.spongepowered.api.event.Subscribe;
import org.spongepowered.api.event.world.WorldLoadEvent;
import org.spongepowered.api.plugin.Plugin;

@Plugin(id = "test", name = "Test", version = "1.0")
public class Test {
	@Subscribe
	public void onWorldLoad(WorldLoadEvent e) {
		System.out.println(e.getWorld());
	}
}
```
you get `WorldClient`s printed as well as `WorldServer`s.